wrappers/python/nntile/types.py

```diff
@@ -0,0 +1,45 @@
+# @copyright (c) 2022-present Skolkovo Institute of Science and Technology
+#                              (Skoltech), Russia. All rights reserved.
+#                2023-present Artificial Intelligence Research Institute
+#                              (AIRI), Russia. All rights reserved.
+#
+# NNTile is software framework for fast training of big neural networks on
+# distributed-memory heterogeneous systems based on StarPU runtime system.
+#
+# @file wrappers/python/nntile/tensor.py
+# Multiprecision tensor with operations
+#
+# @version 1.0.0
+
+from .nntile_core import tensor as core_tensor
+
+# Multiprecision tensor as a union type for all precisions
+Tensor = core_tensor.Tensor_fp32 | core_tensor.Tensor_fp64
+# Optional tensor argument
+TensorOrNone = Tensor | None
+# Union of multiprecision tensor and float
+TensorOrFloat = Tensor | float
+TensorFloatOrInt = Tensor | core_tensor.Tensor_int64
+
+# Struct meant for tensor, its gradient and a flag if gradient is required
+class TensorMoments(object):
+    value: TensorOrNone
+    grad: TensorOrNone
+    grad_required: bool
```

Comment:
FYI

- These lines do nothing except binding attributes to specific types. They are just annotations.
- But this annotations can be used by `dataclasses` or `pydantic` via `__init_subclass__` method or explicit class decoration.
- There is not need to use `object` as an inheritance base since Python 3.

Reply:
Sorry, but this is out of context of this PR. I just split already existed code. If you want, we can discuss it separately